android-client-sdk/src/main/java/com/devcycle/sdk/android/api/DVCClient.kt

```diff
@@ -197,6 +197,7 @@ class DVCClient private constructor(
     @JvmOverloads
     @Synchronized
     fun resetUser(callback: DVCCallback<Map<String, Variable<Any>>>? = null) {
+        clearAnonUserId()
```

Comment:
if the user is anonymous and calls resetUser, wouldn't we be using the wrong anonymous user id if the reset call fails?

Reply:
due to lack of sessions, does that matter?

Reply:
it could set it in a weird state where it doesn't have the anon user id saved but the new request also failed

Reply:
updated to get the anonUserId before clearing, so that if the reset fails we can set it back to the original value

Reply:
why not just clear it after the call succeeds?